### PR TITLE
os/board/rtl8721csm : add fault prevention check

### PR DESCRIPTION
**Pull request by keerthanalex**
add lock to prevent from accessing 0 address
add data length check to avoid hardfault
fix hardfault caused by timer task preempt wifi driver task

### CODE REVIEW

**Review by j11591 — DISMISSED** [review]
LGTM

File: os/board/rtl8721csm/src/component/common/api/wifi/wifi_conf.c [hunk 1]
```diff
@@ -1980,6 +1980,7 @@ int wifi_scan_networks_with_ssid(int (results_handler)(char*buf, int buflen, cha
 	scan_buf_arg scan_buf;
 	int ret;
 
+	scan_buf.buf_len = 0;
```
Comment (jeongchanKim) [inline os/board/rtl8721csm/src/component/common/api/wifi/wifi_conf.c:1983]: 'SCAN_BUF.BUF_LEN' is assigned in the 1984 line, the next line.
Is there any reason to set it as 0 from here?

**Review by j11591 — APPROVED** [review]
LGTM

File: os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c [hunk 2]
```diff
@@ -932,83 +934,106 @@ u32 _tizenrt_timerDelete(_timerHandle xTimer, osdepTickType xBlockTime)
...
 
 void *_tizenrt_timerGetID(_timerHandle xTimer)
 {
 	struct timer_list_priv *timer = (struct timer_list_priv *)xTimer;
-	return timer->data;
+	if (timer != NULL) {
+		return timer->data;
+	} else {
+		DBG_ERR("timer is NULL\n");
+		return 0;	/*No data when timer is NULL*/
```
[the remark this answers is not in the document]
Comment (keerthanalex) [reply os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c]: @sunghan-chang modified as suggested. Please help to review.

**Review by sunghan-chang — APPROVED** [review]

**Review by ewoodev — APPROVED** [review]
LGTM~

**Review by jylee9613 — DISMISSED** [review]
LGTM

**Comment by keerthanalex** [comment]
> @keerthanalex Because PR 5813 was merged, there are some conflicts. Could you resolve them?

@sunghan-chang Since 5813 already includes wlan lib changes, this PR has been updated with other files after rebase.

**Review by j11591 — APPROVED** [review]
LGTM

**Pull request merged**